src/script/auth/page/ConversationJoin.tsx

```diff
@@ -59,124 +59,91 @@ import {parseError, parseValidationErrors} from '../util/errorUtil';
 import * as StringUtil from '../util/stringUtil';
 import {getURLParameter, hasURLParameter, pathWithParams} from '../util/urlUtil';
 
-interface Props extends React.HTMLProps<HTMLDivElement>, RouteComponentProps {}
+interface Props extends React.HTMLProps<HTMLDivElement> {}
 
-interface State {
-  accentColor: AccentColor.AccentColor;
-  conversationCode: string;
-  conversationKey: string;
-  enteredName: string;
-  error: Error;
-  expiresIn: number;
-  forceNewTemporaryGuestAccount: boolean;
-  isValidLink: boolean;
-  isValidName: boolean;
-  showCookiePolicyBanner: boolean;
-}
+const ConversationJoin = ({
+  doCheckConversationCode,
+  doJoinConversationByCode,
+  doInit,
+  doRegisterWireless,
+  setLastEventDate,
+  doLogout,
+  isAuthenticated,
+  isTemporaryGuest,
+  selfName,
+  conversationError,
+}: Props & ConnectedProps & DispatchProps) => {
+  const nameInput = React.useRef<HTMLInputElement>();
+  const {formatMessage: _} = useIntl();
 
-type CombinedProps = Props & ConnectedProps & DispatchProps & InjectedIntlProps;
+  const [accentColor] = useState(AccentColor.random());
+  const [isPwaEnabled, setIsPwaEnabled] = useState();
+  const [conversationCode, setConversationCode] = useState();
+  const [conversationKey, setConversationKey] = useState();
+  const [enteredName, setEnteredName] = useState();
+  const [error, setError] = useState();
+  const [expiresIn, setExpiresIn] = useState();
+  const [forceNewTemporaryGuestAccount, setForceNewTemporaryGuestAccount] = useState(false);
+  const [isValidLink, setIsValidLink] = useState(true);
+  const [isValidName, setIsValidName] = useState(true);
+  const [showCookiePolicyBanner, setShowCookiePolicyBanner] = useState(true);
 
-class ConversationJoin extends React.Component<CombinedProps, State> {
-  nameInput: React.RefObject<any> = React.createRef();
-  state: State = {
-    accentColor: AccentColor.random(),
-    conversationCode: null,
-    conversationKey: null,
-    enteredName: '',
-    error: null,
-    expiresIn: undefined,
-    forceNewTemporaryGuestAccount: false,
-    isValidLink: true,
-    isValidName: true,
-    showCookiePolicyBanner: true,
-  };
-
-  readAndUpdateParamsFromUrl = (nextProps: CombinedProps) => {
-    if (this.isPwaEnabled()) {
-      this.setState({forceNewTemporaryGuestAccount: true});
-    }
-    const conversationCode = getURLParameter(QUERY_KEY.CONVERSATION_CODE);
-    const conversationKey = getURLParameter(QUERY_KEY.CONVERSATION_KEY);
-    const expiresIn = parseInt(getURLParameter(QUERY_KEY.JOIN_EXPIRES), 10) || undefined;
+  useEffect(() => {
```

Comment:
This might cause issues, as the original lifecycle event was `componentWillReceiveProps`. This useEffect will now only trigger once in the beginning as it has the second parameter set to `[]`;